src/compiler/interface.jl

```diff
@@ -4,11 +4,11 @@ using Core: Typeof
 import Base: copy!, IdSet
 import Base.Broadcast: broadcasted, materialize!
 
-mutable struct Context <: AContext
+mutable struct Context{I} <: AContext
```

Comment:
What is this I for? Can you add some documentation comments?

Reply:
I guess it stands for "implicit"

Reply:
Added a small explainer, see if that looks good to you :)